lib/timers.js

```diff
@@ -418,6 +404,27 @@ function processImmediate() {
 }
 
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw === true) {
```

Comment:
If you return early in a `finally` block, the exception wouldn't be thrown, which would change behavior.